components/infobox/wikis/mobilelegends/infobox_hero_custom.lua

```diff
@@ -0,0 +1,180 @@
+
+	unit.getWikiCategories = CustomHero.getWikiCategories
+	unit.setLpdbData = CustomHero.setLpdbData
+	unit.createWidgetInjector = CustomHero.createWidgetInjector
+
+	return unit:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells()
+	local widgets = {
+		Cell{name = 'Resource Bar', content = {_args.secondarybar}},
+		Cell{name = 'Secondary Bar', content = {_args.secondarybar1}},
+		Cell{name = 'Secondary Attributes', content = {_args.secondaryattributes1}},
+		Cell{name = 'Release Date', content = {_args.releasedate}},
+		Cell{name = 'Species', content = {_args.species}},
+		Cell{name = 'Year of birth', content = {_args.birth}},
+		Cell{name = 'Faction(s)', content = {_args.factions}},
+	}
+
+	if not (
+		String.isEmpty(_args.hp) and
+		String.isEmpty(_args.hplvl) and
+		String.isEmpty(_args.hpreg) and
+		String.isEmpty(_args.hpreglvl)
+	) then
+		table.insert(widgets, Title{name = 'Base Statistics'})
+	end
+
+	table.insert(widgets, Cell{name = 'Health', content = {_args.hp}})
+	table.insert(widgets, Cell{name = 'Health Regen', content = {_args.hpreg}})
+	table.insert(widgets, Cell{name = 'Mana', content = {_args.mana}})
+	table.insert(widgets, Cell{name = 'Mana Regen', content = {_args.manareg}})
+	table.insert(widgets, Cell{name = 'Cooldown Reduction', content = {_args.cdr}})
+	table.insert(widgets, Cell{name = 'Energy', content = {_args.energy}})
+	table.insert(widgets, Cell{name = 'Energy Regen', content = {_args.energyreg}})
+	table.insert(widgets, Cell{name = 'Attack Type', content = {_args.attacktype}})
+	table.insert(widgets, Cell{name = 'Attack Damage', content = {_args.damage}})
+	table.insert(widgets, Cell{name = 'Attack Speed', content = {_args.attackspeed}})
+	table.insert(widgets, Cell{name = 'Attack Range', content = {_args.attackrange}})
+	table.insert(widgets, Cell{name = 'Ability Power', content = {_args.ap}})
+	table.insert(widgets, Cell{name = 'Armor', content = {_args.armor}})
+	table.insert(widgets, Cell{name = 'Magic Resistance', content = {_args.magicresistance}})
+	table.insert(widgets, Cell{name = 'Movement Speed', content = {_args.movespeed}})
+	table.insert(widgets, Title{name = 'Esports Statistics'})
+
+	local stats = HeroWL.create({hero = _args.heroname or _pagename})
+	stats = mw.text.split(stats, ';')
+	local winPercentage = (tonumber(stats[1]) or 0) / ((tonumber(stats[1]) or 0) + (tonumber(stats[2]) or 1))
+	winPercentage = Math.round({winPercentage, 4}) * 100
+	local statsDisplay = (stats[1] or 0) .. 'W : ' .. (stats[2] or 0) .. 'L (' .. winPercentage .. '%)'
+	table.insert(widgets, Cell{name = 'Win Rate', content = {statsDisplay}})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'header' then
+		return {
+			Header{
+				name = _args.heroname,
+				subHeader = _args.title,
+				image = _args.image,
+				imageDefault = _args.default,
+				imageDark = _args.imagedark or _args.imagedarkmode,
+				imageDefaultDark = _args.defaultdark or _args.defaultdarkmode,
```

Comment:
Feels like basic image handling should be done in commons, not customs

Reply:
the subheader is what needs the customization here

Reply:
Override Unit:nameDisplay and add in a overridable function for the subHeader?